### PR TITLE
🌱 Add BMO 0.8 e2e

### PR DESCRIPTION
**Pull request by tuminoid**
Add BMO 0.8 e2e and drop BMO 0.4. Add ironic 25.0 and 26.0 to overlays.

### CODE REVIEW

**Comment by mquhuy** [comment]
We can remove the ironic upgrade tests that runs on release-0.6 (L63-80 on `ironic.yaml`). Also remove `ironic-deployment/overlays/e2e-release-24.0-with-inspector` as it's no longer in use.

**Comment by mquhuy** [comment]
> We can remove the ironic upgrade tests that runs on release-0.6 (L63-80 on `ironic.yaml`). Also remove `ironic-deployment/overlays/e2e-release-24.0-with-inspector` as it's no longer in use.

The `e2e-with-inspector` could be removed as well.

**Comment by tuminoid** [comment]
> > We can remove the ironic upgrade tests that runs on release-0.6 (L63-80 on `ironic.yaml`). Also remove `ironic-deployment/overlays/e2e-release-24.0-with-inspector` as it's no longer in use.
> 
> The `e2e-with-inspector` could be removed as well.

Were still testing 24.0, so why would we remove it?

**Comment by mquhuy** [comment]
> > > We can remove the ironic upgrade tests that runs on release-0.6 (L63-80 on `ironic.yaml`). Also remove `ironic-deployment/overlays/e2e-release-24.0-with-inspector` as it's no longer in use.
> > 
> > 
> > The `e2e-with-inspector` could be removed as well.
> 
> Were still testing 24.0, so why would we remove it?

The idea is to test ironic upgrade with BMO version fixed to the latest stable release.
If we test 24.0, then please change the BMO overlay to release-0.8 on that test.

**Comment by tuminoid** [comment]
OK, got the spec meaning now, revamped them. PTAL.

**Comment by mquhuy** [comment]
/lgtm
The cleaning up can be done later then.

File: test/e2e/config/fixture.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ images:
 - name: quay.io/metal3-io/baremetal-operator:e2e
   loadBehavior: tryLoad
 # Save some time and network by using cached images if available
-- name: quay.io/metal3-io/baremetal-operator:release-0.4
+- name: quay.io/metal3-io/baremetal-operator:release-0.8
```
Comment (kashifest) [inline test/e2e/config/fixture.yaml:6]: from release-0.4 to release-0.8 ? Why the jump? Were we doing things wrongly before?

File: test/e2e/config/ironic.yaml [hunk 2]
```diff
@@ -5,7 +5,7 @@ images:
 - name: quay.io/metal3-io/ironic:e2e
   loadBehavior: tryLoad
 # Save some time and network by using cached images if available
-- name: quay.io/metal3-io/baremetal-operator:release-0.4
+- name: quay.io/metal3-io/baremetal-operator:release-0.8
```
Comment (kashifest) [inline test/e2e/config/ironic.yaml:8]: same question here

File: test/e2e/config/fixture.yaml (hunk 1 above)
Comment (tuminoid) [reply test/e2e/config/fixture.yaml:6]: It is preloading images. We don't use 0.4 anymore, and I guess forgot to bump it. I figured 0.8 is most used, so preloading it helps the most?

File: test/e2e/config/ironic.yaml (hunk 2 above)
Comment (tuminoid) [reply test/e2e/config/ironic.yaml:8]: Same as above

**Pull request merged**